components/butterfly/src/server.rs

```diff
@@ -657,14 +644,23 @@ impl Server {
                       member_id,
                       trace_incarnation,
                       trace_health);
-
-            if member_id != self.member_id() && health == Health::Departed {
-                self.rumor_heat.purge(&member_id);
-            }
-            self.rumor_heat.start_hot_rumor(rk);
         }
     }
 
+    /// Mark a service for deletion. This means butterfly will mark the Service, ServiceConfig and
+    /// ServiceFile rumors related to this service as expired. They will eventually be purged.
+    ///
+    /// In addition, Election and ElectionUpdate rumors for this service group are also expired.
+    /// Without the Service* rumors, there's no service group to speak of, so there's no point
+    /// holding onto Election* rumors either.
+    pub fn mark_service_for_deletion(&self, service: &Service) {
+        self.service_store.expire_all_for_key(service.key());
+        self.service_config_store.expire_all_for_key(service.key());
+        self.service_file_store.expire_all_for_key(service.key());
+        self.election_store.expire_all_for_key(service.key());
+        self.update_store.expire_all_for_key(service.key());
```

Comment:
This should only be called when a service is unloaded, not stopped.  It's possible that I got the wrong code path in the supervisor code, or rather, got a code that that is shared by both `stop` and `unload`.  I will double check and test this some more.

Reply:
I think it may be called in either case. In any event, the main point about other Supervisors in the network still needing to access the other rumors still stands, I think.

Reply:
... which raises an interesting point about when those service group-wide rumors (config, file, election, update) should expire, since that should conceptually take place after the last Supervisor running a service in that group unloads that service.

That would take some more logic (not a whole lot more, but a different kind of logic than what's here in this PR currently). Of course, member and service rumors are the majority of our data by a long shot, so that's a relatively small concern (but one we should still tackle at some point, I think).

Reply:
@christophermaier If I'm following you here, I think that means that the expiration date _should not_ be gossiped around the network.  If we gossip the expiration date around the network, as we currently do, then every node will receive it and expire said rumor.  If we don't gossip the expiration date, that means each node will expire things independently of the others, which sounds like what you're describing.

If I'm interpreting your comments correctly, it sounds like only `Member` rumors should have their expiration date gossiped (since by definition, once a `Member` hits `Departed`, the entire network has agreed that that node is gone).

Does this sound right to you?

Incidentally, this feels like it's in direct opposition to the notion that @baumanj were discussing farther up in this PR, around every rumor expiring unless it's refreshed.  It'd be good to reconcile these approaches at some point :joy:

Reply:
Just to follow up on this - I'm going to take another stab at the refresh mechanism that I couldn't get working a few months ago, which will involve removing this particular bit of code.